docs/source/terminology.rst

```diff
@@ -0,0 +1,45 @@
+Terminology
+======================
+
+Dask is a large project with many terms that might be unfamiliar to new users. This glossary is to provide a way
+for new and returning users alike to find the definitions of various Dask terms.
+
+..
+    _ the following terms should be in alphabetical order
```

Comment:
It might be more helpful if we could group terms by concept, rather than alphabetically. Alphabetical order really only helps people know where to find a word on the page, which is also easily achieved with Ctrl+F. Grouping terms by concept has a higher chance of users visiting this page learning something new, in addition to the term they came for.